framework/src/modules/chain/bft/bft.js

```diff
@@ -301,7 +314,30 @@ class BFT extends EventEmitter {
 		rows.forEach(row => {
 			if (row.height !== 1 && row.version !== 2) return;
 
-			this.finalityManager.addBlockHeader(extractBFTBlockHeaderFromBlock(row));
+			const delegate = delegateMinHeightActiveList[row.generatorPublicKey];
+
+			if (!delegate) {
+				throw new Error('Delegate was not found in the active delegate list!');
+			}
+
+			const delegateMinHeightActive = delegate.activeHeights.filter(
```

Comment:
```suggestion
			const [delegateMinHeightActive] = delegate.activeHeights.filter(
```
Instead of using `[0]`